pkg/ocm/ocm.go

```diff
@@ -0,0 +1,133 @@
+	}
+	client.cfg = cfg
+
+	conn, err := client.establishConnection()
+	if err != nil {
+		return client, fmt.Errorf("can't create connection: %w", err) // ConnectionError{client:client, err:err}
+	}
+	client.conn = conn
+
+	return client, nil
+}
+
+// loadConfig loads the configuration file (ocmConfigFile, ~/.ocm.json, /ocm/ocm.json)
+func newConfigFromFile(ocmConfigFile string) (*sdkcfg.Config, error) {
+	if ocmConfigFile != "" {
+		err := os.Setenv("OCM_CONFIG", ocmConfigFile)
+		if err != nil {
+			return nil, err
+		}
+	}
+	// Load the configuration file from std path
+	cfg, err := sdkcfg.Load()
+	if err != nil {
+		return nil, err
+	}
+
+	if cfg == nil {
+		err = fmt.Errorf("not logged in")
+		return nil, err
+	}
+	return cfg, err
+}
+
+// establishConnection creates a connection from the client settings
+func (client ocmclient) establishConnection() (*sdk.Connection, error) {
+
+	if client.cfg == (&sdkcfg.Config{}) {
+		return nil, fmt.Errorf("config is empty, aborting")
+	}
+	builder := sdk.NewConnectionBuilder()
+	if client.cfg.TokenURL != "" {
+		builder.TokenURL(client.cfg.TokenURL)
+	}
+	if client.cfg.ClientID != "" || client.cfg.ClientSecret != "" {
+		builder.Client(client.cfg.ClientID, client.cfg.ClientSecret)
+	}
+	if client.cfg.Scopes != nil {
+		builder.Scopes(client.cfg.Scopes...)
+	}
+	if client.cfg.URL != "" {
+		builder.URL(client.cfg.URL)
+	}
+	if client.cfg.User != "" || client.cfg.Password != "" {
+		builder.User(client.cfg.User, client.cfg.Password)
+	}
+	if client.cfg.AccessToken != "" {
+		builder.Tokens(client.cfg.AccessToken)
+	}
+	if client.cfg.RefreshToken != "" {
+		builder.Tokens(client.cfg.RefreshToken)
+	}
+	builder.Insecure(client.cfg.Insecure)
+	// Create the connection:
+	conn, err := builder.Build()
+	if err != nil {
+		return nil, err
+	}
+	return conn, nil
+}
+
+// GetAWSAccountClaim gets the AWS Account Claim object for a given cluster
+func (client ocmclient) GetAWSAccountClaim(clusterID string) (*awsv1alpha1.AccountClaim, error) {
+	ac := &awsv1alpha1.AccountClaim{}
+	acString, err := client.getClusterResource(clusterID, "aws_account_claim")
+	if err != nil {
+		return ac, err
+	}
+	err = json.Unmarshal([]byte(acString), ac)
+	return ac, err
+}
+
+// getClusterResource handles caching the live cluster resource endpoint and returns the key
+// you're looking for. This function only caches certain values, as some are not necessary at
+// this time, such as the cluster install logs.
+func (client ocmclient) getClusterResource(clusterID string, resourceKey string) (string, error) {
+
+	response, err := client.conn.ClustersMgmt().V1().Clusters().Cluster(clusterID).Resources().Live().Get().Send()
+	if err != nil {
+		return "", err
+	}
+	return response.Body().Resources()[resourceKey], nil
+}
+
+// GetClusterDeployment gets the ClusterDeployment object for a given cluster
+func (client ocmclient) GetClusterDeployment(clusterID string) (*hivev1.ClusterDeployment, error) {
```

Comment:
This PR looks pretty good. Was this function and GetAWSAccountClaim function tested with your credentials? Just want to confirm they work without hive connection ?

Reply:
When I paired with @ramonbutter , I saw the command suceed